crm_rma_location/models/stock_warehouse.py

```diff
@@ -0,0 +1,180 @@
+        # each warehouse: reception, internal, pick, pack, ship.
+        max_sequence = self.env['stock.picking.type'].\
+            search_read([], ['sequence'], order='sequence desc')
+        max_sequence = max_sequence and max_sequence[0]['sequence'] or 0
+
+        in_type_id = warehouse.rma_in_type_id
+        if not in_type_id:
+            in_type_id = picking_type_obj.create(vals={
+                'name': _('RMA Receipts'),
+                'warehouse_id': warehouse.id,
+                'code': 'incoming',
+                'sequence_id': in_seq_id.id,
+                'default_location_src_id': customer_loc.id,
+                'default_location_dest_id': wh_stock_loc.id,
+                'sequence': max_sequence + 4,
+                'color': color})
+
+        out_type_id = warehouse.rma_out_type_id
+        if not out_type_id:
+            out_type_id = picking_type_obj.create(vals={
+                'name': _('RMA Delivery Orders'),
+                'warehouse_id': warehouse.id,
+                'code': 'outgoing',
+                'sequence_id': out_seq_id.id,
+                'return_picking_type_id': in_type_id.id,
+                'default_location_src_id': wh_stock_loc.id,
+                'default_location_dest_id': supplier_loc.id,
+                'sequence': max_sequence + 1,
+                'color': color})
+            in_type_id.write({'return_picking_type_id': out_type_id.id})
+
+        int_type_id = warehouse.rma_int_type_id
+        if not int_type_id:
+            int_type_id = picking_type_obj.create(vals={
+                'name': _('RMA Internal Transfers'),
+                'warehouse_id': warehouse.id,
+                'code': 'internal',
+                'sequence_id': int_seq_id.id,
+                'default_location_src_id': wh_stock_loc.id,
+                'default_location_dest_id': wh_stock_loc.id,
+                'active': True,
+                'sequence': max_sequence + 2,
+                'color': color})
+
+        # write picking types on WH
+        vals = {
+            'rma_in_type_id': in_type_id.id,
+            'rma_out_type_id': out_type_id.id,
+            'rma_int_type_id': int_type_id.id,
+        }
+        return vals
+
+    @api.model
+    def create_locations_rma(self, warehouse_id):
+        """
+        Create a RMA location for RMA movements that takes place when internal,
+        outgoing or incoming pickings are made from/to this location
+        """
+        vals = {}
+
+        location_obj = self.env['stock.location']
+        context_with_inactive = self.env.context.copy()
+        context_with_inactive['active_test'] = False
+        view_location_id = warehouse_id.view_location_id.id
+
+        if not warehouse_id.lot_rma_id:
+            loc_vals = {
+                'name': _('RMA'),
+                'usage': 'internal',
+                'location_id': view_location_id,
+                'active': True,
+            }
+            if vals.get('company_id'):
```

Comment:
This is still never reached, because vals was initialized to {} and never change. You have to use warehouse_id.company_id